tests/unit/test_spaces.cpp

```diff
@@ -115,25 +135,21 @@ TEST_P(FP32SpacesOptimizationTest, FP32L2SqrTest) {
 
     dist_func_t<float> arch_opt_func;
     float baseline = FP32_L2Sqr(v, v2, dim);
-    switch (optimization) {
-    case ARCH_OPT_AVX512_BW_VBMI2:
-    case ARCH_OPT_AVX512_F:
-        arch_opt_func = L2_FP32_GetDistFunc(dim, ARCH_OPT_AVX512_F);
+    if (optimization.features.avx512f) {
```

Comment:
either check also the compiler flags or add a comment that we assume that the compiler supports the highest arch optimization